Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1

```diff
@@ -386,6 +386,174 @@ function Publish-TerminologyDacpac() {
+
+    try {
+        $response = Invoke-RestMethod -Uri $dpsUrl/ExecuteDataMart -Method POST -UseDefaultCredentials -Headers $headers -Body "{ `"DataMartId`": $dataMartId, `"BatchExecution`": { `"PipelineType`": `"Migration`", `"OverrideLoadType`": `"Incremental`", `"LoggingLevel`": `"Diagnostic`" } }"
+        $batchExecutionId = ($response.value | ConvertFrom-Json).Id
+        Write-DosMessage -Level "Info" -Message  "Batch execution successfully sent to the data processing service."
+
+        return $batchExecutionId;
+    }
+    catch {
+        Write-DosMessage -Level "Error" -Message  "POST for '$name' failed with status code:" $_.Exception.Response.StatusCode.value__ 
+        Write-DosMessage -Level "Error" -Message  "Description:" $_.Exception.Response.StatusDescription
+    }
+}
+
+function Poll-BatchExecutions {
+   param(
+        [string] $terminologyBatchExecutionId = $(throw "Please specify the terminologyBatchExecutionId"),
+        [string] $sharedTerminologyBatchExecutionId = $(throw "Please specify the sharedTerminologyBatchExecutionId")
+        )
+
+    $dpsUrl = Get-ServiceFromDiscovery -name "DataProcessingService" -version 1
+    $terminologyUrl = "$dpsUrl/BatchExecutions($terminologyBatchExecutionId)"
+    $terminologyResponse = "";
+    $sharedTerminologyResponse = "";
+    $wasSuccessful = 0;
+
+    foreach($i in 1..30) {
+        Start-Sleep -s 5
+        if($terminologyResponse -ne "Failed") {
+            $terminologyResponse = Invoke-RestMethod -Uri $terminologyUrl -Method GET -UseDefaultCredentials
+        }
+        if($sharedTerminologyResponse -ne "Failed") {
+            $sharedTerminologyResponse = Invoke-RestMethod -Uri $terminologyUrl -Method GET -UseDefaultCredentials
+        }
+
+        if($terminologyResponse.Status -eq "Failed" -or $sharedTerminologyResponse.Status -eq "Failed") {
+            Write-DosMessage -Level "Error" -Error "One of the batch executions has failied. Please check EDW Console for additional logging. Halting the terminology registration"
+            break
+        }
+        if($terminologyResponse.Status -eq "Success" -and $sharedTerminologyResponse.Status -eq "Success") {
+            Write-DosMessage -Level "Error" -Message "Both batch executions have successfully been run."
+            $wasSuccessful = 1;
+            break
+        }
+
+        Write-DosMessage -Level "Information" -Message  "Terminology status: $($terminologyResponse.Status)"
+        Write-DosMessage -Level "Information" -Message  "SharedTerminology status: $($sharedTerminologyResponse.Status)"
+    }
+    return $wasSuccessful;
+
+}
+
+
+function Add-MetadataAndStructures() {
+    param(
+        [PSCumstonObject] $Config
+    )
+    $discoveryServiceUrl = $Config.discoveryServiceUrl
+    $metadataDatabase = $Config.metadataDatabaseName
+    $sqlAddress = $Config.sqlAddress
+      
+   
+    # get IdentityID from IdentityBASE
```

Comment:
I am really confused why you get IdentityID, the only place you use it is to delete the record again... were you trying to give permissions through `IdentityRoleBASE`? I dont see where that was done